src/android/Sms.java

```diff
@@ -55,6 +57,11 @@ else if (action.equals(ACTION_HAS_PERMISSION)) {
 			callbackContext.sendPluginResult(new PluginResult(PluginResult.Status.OK, hasPermission()));
 			return true;
 		}
+		else if (action.equals(ACTION_REQUEST_PERMISSION)) {
+			requestPermission();
+			callbackContext.sendPluginResult(new PluginResult(PluginResult.Status.OK, true));
```

Comment:
Shouldn't we return `hasPermission()` instead of `true`? If `requestPermission()` is a blocking method, it might be useful to return `false` if the user denied the access.

Reply:
Hi @dbaq. That is a good idea. I tried to change it but as `requestPermission()` is non blocking, `hasPermission()` will be called before user accepts or declines, resulting in false being returned every time.

```java
else if (action.equals(ACTION_REQUEST_PERMISSION)) {
  requestPermission();
  callbackContext.sendPluginResult(new PluginResult(PluginResult.Status.OK, hasPermission()));
  return true;
}
```

Do you have another solution?

Reply:
That's what I thought, there is a solution I believe. When you call `requestPermission()`, the method `onRequestPermissionResult` is executed after the user selected deny/allow. What we currently do is, if the user allows the permission we call `sendSMS` (https://github.com/gotling/cordova-sms-plugin/blob/30d48efd7d8ab5757a818547c418fa4014c5c0cf/src/android/Sms.java#L83).

So in the case we just want to request the permission, we need to call `callbackContext.sendPluginResult(new PluginResult(PluginResult.Status.OK, true));` instead of `sendSMS` in the `onRequestPermissionResult`.

To differentiate `sensSMS` vs `requestPermission`, you have the `requestCode` parameter:
```java
public void onRequestPermissionResult(int requestCode, String[] permissions, int[] grantResults) throws JSONException {
```

You can simply add:
```java
private static final int REQUEST_PERMISSION_REQ_CODE = 1;
```

You also need to add this paramete to `requestPermission` so you can identify the call.
 (`requestPermission(REQUEST_PERMISSION_REQ_CODE)` and `requestPermission(SEND_SMS_REQ_CODE)`).

To finish, you need to either call `sendSMS()` if the `requestCode` equals `SEND_SMS_REQ_CODE` or ```java
  callbackContext.sendPluginResult(new PluginResult(PluginResult.Status.OK, hasPermission()));
return;
```

Also, make sure to remove that line from your previous snippet:
```java
  callbackContext.sendPluginResult(new PluginResult(PluginResult.Status.OK, hasPermission()));
```

Hopefully, I am clear enough. That should work perfectly.

Reply:
And make sure to use the new param when you call:
```java
	private void requestPermission(int requestCode) {
		cordova.requestPermission(this, requestCode, android.Manifest.permission.SEND_SMS);
	}
```